### ISSUE TITLE
Add instrumentation metrics to Smart Search

### ISSUE THREAD

**Issue by almeidapaulooliveira** [sourcegraph/sourcegraph#45188]
### Related design issue:
- #39771

### Description
With Smart Search, we're curious if visitor behavior changes at all. Is retention better? Do we see a rise in searches? Is the correlation between searches and trial signups changing?

For this, we're fully instrumenting the Smart Search component. More context [here](https://sourcegraph.slack.com/archives/C046X8AG5NY/p1669752471037509).

Figma page with defined events:
https://www.figma.com/file/GikUoxdCrRTmbQpiu72AUi/UX-for-%E2%80%99feeling-lucky%E2%80%99-search-mode.?node-id=3304%3A9738

**Comment by quinnkeast** [comment]
It seems like our guidelines on naming events in Amplitude were lost at some point (@rrhyne or @ebrodymoore any idea what's up there)? Previously [they were listed here](https://handbook.sourcegraph.com/departments/bizops/tools/amplitude/#adding-events-to-amplitude), but I was able to [find them in history](https://sourcegraph.com/github.com/sourcegraph/handbook@c9e28c7c9ecc2c32897e772292e12e629f8b27e7/-/blob/content/departments/bizops/tools/amplitude.md#adding-events-to-amplitude).

TL;DR:

>We use [object action framework in Proper Case](https://segment.com/academy/collecting-data/naming-conventions-for-clean-data/) for our naming. For example, in eventLogger this would show up as 'SearchSubmitted', and in Amplitude it's converted to 'Search Submitted'.

>Name space your event names so events in a group are easily recognized. For example: ClickedCTA follws the convension, but does not help users distinguish which CTA or where the event occurred. PostSignUpFlowCLickedBextCTA helps identify this CTA was for the browser extension.

Based on these guidelines, the event names defined here should be tweaked a bit like:

`ClickedOnHideSmartSearchQueries` → `SmartSearchHideClicked`
`ClickedOnSmartSearchQuery` → `SmartSearchQueryClicked`
etc.

I'd also suggest changing `ClickedOnSmartSearchDropdwon` from "Dropdown" to "Toggle." The dropdown is what's presented, but it's a toggle button used as a trigger. So, that'd be `ClickedOnSmartSearchDropdwon` → `SmartSearchToggleClicked`.

With these changes across the board, LGTM!

**Comment by almeidapaulooliveira** [comment]
Fixed. Thanks, Quinn.
